nightfall-optimist/src/services/block-assembler.mjs

```diff
@@ -117,49 +126,56 @@ export async function conditionalMakeBlock(proposer) {
         });
 
         // propose this block to the Shield contract here
-        const unsignedProposeBlockTransaction = await (
-          await waitForContract(STATE_CONTRACT_NAME)
-        ).methods
+        const txDataToSign = await stateContractInstance.methods
           .proposeBlock(
             Block.buildSolidityStruct(block),
             transactions.map(t => Transaction.buildSolidityStruct(t)),
           )
           .encodeABI();
+        const { address } = web3.eth.accounts.privateKeyToAccount(ethPrivateKey);
```

Comment:
if we are fixing the proposer private key to the one in config, why do we need to compute the Ethereum address in every block? Or why cant we pass it as a parameter as well?

Reply:
I will think of the best way to improve this depending on your answers above re Eth address and private key :pray: Thanks

Reply:
I’m talking about efficiency point of view. I am just wondering why we need to compute something that we have already computed in app.mjs